File: Rust/rust_project/src/array.rs
use rand;

pub fn random_access(nums: &[i32]) -> i32 {
    let random_index = rand::random_range(0..(nums.len() - 1));
    let random_num = nums[random_index];
    random_num
}

pub fn insert(arr: &mut [i32], index: usize, value: i32) {
    for i in (index + 1..arr.len()).rev() {
        arr[i] = arr[i - 1];
    }
    arr[index] = value;
}

pub fn remove(arr: &mut [i32], index: usize) {
    for i in index..arr.len() - 1 {
        arr[i] = arr[i + 1];
    }
}

pub fn find(arr: &[i32], target: i32) -> Option<usize> {
    for i in 0..arr.len() {
        
        if arr[i] == target {
            return Some(i);
        }
    }
    None
}

pub fn extend(arr: &[i32], enlarge: usize) -> Vec<i32> {
    let mut res: Vec<i32> = vec![0; arr.len() + enlarge];
    for i in 0..arr.len() {
        res[i] = arr[i];
    }
    
    res
}

File: Java/src/SeqList/Array.java
package SeqList;

import java.util.concurrent.ThreadLocalRandom;

public class Array {
    public int random_access(int[] arr) {
        int randomIndex = ThreadLocalRandom.current().nextInt(0, arr.length);

        return arr[randomIndex];
    }

    public void insert(int[] arr, int num, int index) {
        for (int i = arr.length - 1; i > index; i--) {
            arr[i] = arr[i - 1];
        }
        arr[index] = num;
    }

    public void remove(int[] arr, int index) {
        for (int i = index; i < arr.length - 1; i++) {
            arr[i] = arr[i + 1];
        }
    }

    public int find(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }

    public int[] extend(int[] arr, int enlarge) {
        int[] res = new int[arr.length + enlarge];
        System.arraycopy(arr, 0, res, 0, arr.length);
        return res;
    }
}
